cmd/rcd/rcd.go

```diff
@@ -32,18 +41,209 @@ See the [rc documentation](/rc/) for more info on the rc flags.
 		}
 		if s == nil {
 			log.Fatal("rc server not configured")
 		}
+
 		s.Wait()
 	},
 }
+
+/*
+GetLatestReleaseURL returns the latest release details of the rclone-webui-react
+*/
+func GetLatestReleaseURL() (string, string, int) {
+	resp, err := http.Get("https://api.github.com/repos/negative0/rclone-webui-react/releases/latest")
+	if err != nil {
+		panic("Error getting latest release of rclone-webui")
+	}
+	results := GitHubRequest{}
+	if err := json.NewDecoder(resp.Body).Decode(&results); err != nil {
+		panic("Could not decode results from http request")
+	}
+
+	res := results.Assets[0].BrowserDownloadURL
+	tag := results.TagName
+	size := results.Assets[0].Size
+	//fmt.Println( "URL:" + res)
+
+	return res, tag, size
+
+}
+
+/*
+DownloadFile is a helper function to download a file from url to the filepath
+*/
+func DownloadFile(filepath string, url string) error {
+
+	// Get the data
+	resp, err := http.Get(url)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := resp.Body.Close(); err != nil {
+			panic(err)
+		}
+	}()
+
+	// Create the file
+	out, err := os.Create(filepath)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := out.Close(); err != nil {
+			panic(err)
+		}
+	}()
+
+	// Write the body to file
+	_, err = io.Copy(out, resp.Body)
+	return err
+}
+
+/*
+Unzip is a helper function to unzip a file specified in src to path dest
+*/
+func Unzip(src, dest string) error {
+	r, err := zip.OpenReader(src)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := r.Close(); err != nil {
+			panic(err)
+		}
+	}()
+
+	if err := os.MkdirAll(dest, 0755); err != nil {
+		return err
+	}
+
+	// Closure to address file descriptors issue with all the deferred .Close() methods
+	extractAndWriteFile := func(f *zip.File) error {
+		rc, err := f.Open()
+		if err != nil {
+			return err
+		}
+		defer func() {
+			if err := rc.Close(); err != nil {
+				panic(err)
+			}
+		}()
+
+		path := filepath.Join(dest, f.Name)
+
+		if f.FileInfo().IsDir() {
+			if err := os.MkdirAll(path, f.Mode()); err != nil {
+				return err
+			}
+		} else {
+			if err := os.MkdirAll(filepath.Dir(path), f.Mode()); err != nil {
+				return err
+			}
+			f, err := os.OpenFile(path, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, f.Mode())
+			if err != nil {
+				return err
+			}
+			defer func() {
+				if err := f.Close(); err != nil {
+					panic(err)
+				}
+			}()
+
+			_, err = io.Copy(f, rc)
+			if err != nil {
+				return err
+			}
+		}
+		return nil
+	}
+
+	for _, f := range r.File {
+		err := extractAndWriteFile(f)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// exists returns whether the given file or directory exists
+func exists(path string) bool {
+	_, err := os.Stat(path)
+	if err == nil {
+		return true
+	}
+	if os.IsNotExist(err) {
+		return false
+	}
+	return true
+}
+
+/*
+GitHubRequest Maps the GitHub API request to structure
+*/
+type GitHubRequest struct {
```

Comment:
Did you know that this is in the rclone codebase already!  In [get-github-release.go](https://github.com/rclone/rclone/blob/master/bin/get-github-release.go)!

I think eventually we will move the release zips to `rclone.org` so don't worry too much about it for the moment